src/S3/MultipartCopy.php

```diff
@@ -128,15 +129,17 @@ private function createPart($partNumber, $partsCount)
             $data[$k] = $v;
         }
 
-        list($bucket, $key) = explode('/', ltrim($this->source, '/'), 2);
-        $data['CopySource'] = '/' . $bucket . '/' . implode(
-            '/',
-            array_map(
-                'urlencode',
-                explode('/', rawurldecode($key))
-            )
-        );
+        if (is_array($this->source)) {
+            $key = str_replace('?', '%3F', $this->source['source_key']);
+            $data['CopySource'] = '/' . $this->source['source_bucket'] . '/' . $key;
+        } else {
+            list($bucket, $key) = explode('/', ltrim($this->source, '/'), 2);
+            $data['CopySource'] = '/' . $bucket . '/' . $key;
+        }
         $data['PartNumber'] = $partNumber;
+        if (!empty($this->sourceVersionId)) {
```

Comment:
Doesn't look like `sourceVersionId` can be set at all. I'd recommend removing this for now and putting it in to a separate PR so the changes can be reviewed in isolation.

Reply:
What do you mean?  sourceVersionId is set and used directly below, and is set on line 190- it's only needed on the headobject and used here